pkg/mic/mic.go

```diff
@@ -730,7 +736,41 @@ func (c *Client) updateUserMSI(newAssignedIDs []aadpodid.AzureAssignedIdentity,
 
 	node, err := c.NodeClient.Get(nodeName)
 	if err != nil {
-		glog.Errorf("Lookup of identities on node %s resulted in error %v", nodeName, err)
+		glog.Warningf("Unable to get node %s while updating user msis. Error %v", nodeName, err)
+
+		if !strings.Contains(err.Error(), "not found") {
+			return
+		}
+
+		// node is no longer found in the cluster, all the assigned identities that were created in this sync loop
```

Comment:
Can we move this to a utility function and then add UTs.

Reply:
I've added a unit test in MIC to test this specific scenario.